src/Schema/MySQLSchemaManager.php

```diff
@@ -405,15 +405,23 @@ protected function selectTableNames(string $databaseName): Result
 
     protected function selectTableColumns(string $databaseName, ?string $tableName = null): Result
     {
+        // Inverse the aliasing of JSON type to LONGTEXT by MariaDb.
+        if ($this->_platform instanceof MariaDb1027Platform) {
+            [$columnTypeSQL, $joinCheckConstraintSQL] = $this->_platform->getInverseJsonToLongtextAliasSQL();
+        } else {
+            $columnTypeSQL          = 'c.COLUMN_TYPE';
+            $joinCheckConstraintSQL = '';
+        }
```

Comment:
Is this `instanceof` check necessary? Can't we make sure that the method we call here is always present on the platform?

Reply:
You're correct. The check is unnecessary. I included it for clarity as the un-aliasing is relevant only to MariaDb platforms. There are three ways of doing this:

1. as per the patch above
2. add a default `getInverseJsonToLongtextAliasSQL()` to the AbstractMySqlPlatform (returning the `else` part above) and remove the platform check (recognising that this method will be pretty meaningless to MySQL in general), or
3. rely on the MariaDb specific executable comments in the SQL this method generates and move the `getInverseJsonToLongtextAliasSQL()` method from the MariaDbPlatform to the AbstractMySqlPlatform (again recognising this method makes little sense to MySQL in general). 

Which do you prefer?

Reply:
Option 2 was what I had in mind.

Reply:
Great. I'll update the patch and rename the method so it's slightly less meaningless to MySQL in general.